### PR TITLE
Note domain

### PR DESCRIPTION
**Pull request by pushpalata06**
add input validation for create note domain

@kalehrishi

### CODE REVIEW

File: app/Domain/Note.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+use Notes\Model\NoteTag as NoteTagModel;
+
+class Note
+{
+    public function __construct()
+    {
+        $this->validator = new InputValidator();
+    }
+    
+    public function create(NoteModel $noteModel, UserTagModel $userTagModel, NoteTagModel $noteTagModel)
```
Comment (kalehrishi) [inline app/Domain/Note.php]: tags can be multiple.

File: app/Domain/Note.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            $resultsetNoteCreateModel = array(
+                $resultsetNoteDomain,
+                $resultsetUserTagDomain,
+                $resultsetNotTagDomain
+            );
+            return $resultsetNoteCreateModel;
+        }
+    }
+    
+    public function delete(NoteModel $noteModel, NoteTagModel $noteTagModel)
```
Comment (kalehrishi) [inline app/Domain/Note.php]: while deleting note we  have to delete note and not its tag.

**Comment by pushpalata06** [comment]
@amitheda

**Comment by pushpalata06** [comment]
@anushamh 
@gauribhapkar

File: app/Model/Note.php [hunk 3]
```diff
@@ -11,12 +11,77 @@ class Note
...
+
+    public function getIsDeleted()
+    {
+        return $this->isDeleted;
+    }
+    public function setIsDeleted($isDeleted)
+    {
+        $this->isDeleted = $isDeleted;
+    }
+    /*public function __construct($params)
```
Comment (anushamh) [inline app/Model/Note.php]: never commit the code with comments.remove the commented code

File: tests/Domain/NoteTest.php [hunk 4]
```diff
@@ -0,0 +1,307 @@
...
+            echo "Connection failed: " . $e->getMessage();
+        }
+    }
+    
+    public function getDataSet()
+    {
+        return $this->createXMLDataSet(dirname(__FILE__) . '/_files/domain_note_seed.xml');
+    }
+    
+    public function testCanCreate()
```
Comment (anushamh) [inline tests/Domain/NoteTest.php]: remove the test word add it in comments like  /**
    * @test
    *
    **/
and you can change the name of test as it_should_create_note

File: tests/Domain/NoteTest.php [hunk 5]
```diff
@@ -0,0 +1,307 @@
...
+        $collection = new Collection();
+        $collection->add($noteTagModel1);
+        $this->assertEquals($noteTagModel1, $collection->getRow(0));
+        
+        $collection = new Collection();
+        $collection->add($noteTagModel2);
+        $this->assertEquals($noteTagModel2, $collection->getRow(0));
+    }
+    
+    public function testCanCreatebyPassingEmptyArrayTag()
```
Comment (anushamh) [inline tests/Domain/NoteTest.php]: change the name of the test case.Name of the test case could be,it_should_create_note_by_passing_an_empty_array_tag. remove the test name and add the test word in to comments.

File: app/Model/Note.php [hunk 6]
```diff
@@ -11,12 +11,77 @@ class Note
...
+    public function getId()
+    {
+        return $this->id;
+    }
+    public function setId($id)
+    {
+        $this->id = $id;
+    }
+
+    public function getUserId()
```
Comment (gauribhapkar) [inline app/Model/Note.php]: test name should be  it_should_get_userId

File: app/Mapper/Note.php [hunk 7]
```diff
@@ -32,8 +34,8 @@ public function create(NoteModel $noteModel)
     public function delete(NoteModel $noteModel)
```
Comment (amitheda) [inline app/Mapper/Note.php]: Write a single function for delete and update.

**Comment by pushpalata06** [comment]
@kalehrishi 
@amitheda 
@anushamh 
@gauribhapkar

File: app/Domain/Note.php [hunk 8]
```diff
@@ -0,0 +1,119 @@
+<?php
+namespace Notes\Domain;
+
+use Notes\Mapper\Note as NoteMapper;
+use Notes\Mapper\FindNotes as FindNotesMapper;
```
Comment (kalehrishi) [inline app/Domain/Note.php]: Class name should not contain verb.

File: app/Domain/Note.php [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    
+    public function create(NoteModel $noteModel, $tagsInput = null)
+    {
+        if ($this->validator->notNull($noteModel->getTitle())) {
+            $userModel  = new UserModel();
+            $userModel->setId($noteModel->getUserId());
+            
+            $userDomain = new UserDomain();
+            $resultsetUserModel = $userDomain->read($userModel);
+            $noteModel->setUserId($resultsetUserModel->getId());
```
Comment (kalehrishi) [inline app/Domain/Note.php]: UserId validation should be done in Service.

File: app/Domain/Note.php [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+class Note
+{
+    public function __construct()
+    {
+        $this->validator = new InputValidator();
+    }
+    
+    public function create(NoteModel $noteModel, $tagsInput = null)
+    {
+        if ($this->validator->notNull($noteModel->getTitle())) {
```
Comment (kalehrishi) [inline app/Domain/Note.php]: validate userId not empty.

File: app/Domain/Note.php [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+                    $noteTagModel->setNoteId($resultsetNoteModel->getId());
+                    $noteTagModel->setUserTagId($collectionUserTagModel->getRow($i)->getId());
+                    
+                    
+                    $noteTagDomain = new NoteTagDomain();
+                    $collectionNoteTagModel->add($noteTagDomain->create($noteTagModel));
+                    
+                }
+            } else {
+                return $resultsetNoteModel;
```
Comment (kalehrishi) [inline app/Domain/Note.php]: array("noteModel"=>$resultsetNoteModel,)

File: app/Domain/Note.php [hunk 12]
```diff
@@ -0,0 +1,119 @@
...
+    }
+    
+    public function update(NoteModel $noteModel)
+    {
+        if ($this->validator->notNull($noteModel->getId())
+            && $this->validator->validNumber($noteModel->getId())
+            && $this->validator->notNull($noteModel->getTitle())
+            && $this->validator->notNull($noteModel->getBody())) {
+            $noteMapper = new NoteMapper();
+
```
Comment (kalehrishi) [inline app/Domain/Note.php]: This should also update tags.

**Pull request closed without merging**